reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,136 @@
+                    "cluster-vpc-accepter",
+                ):
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_peering": {
+                            "vpc_name": peering.cluster.name,  # type: ignore[union-attr]
+                            "cidr_block": peering.cluster.network.vpc,  # type: ignore[union-attr]
+                        },
+                    }
+                    create_vpc_peering_dict(
+                        vpc_peering_info,
+                        peerings_enteries_dict,
+                        cluster.network.vpc,  # type: ignore[union-attr]
+                    )
+        overlaps_peering_entries_dict = find_cidr_duplicates_and_overlap(
+            cluster.name, peerings_enteries_dict
+        )
+        if overlaps_peering_entries_dict is False:
+            return False
+    return True
+
+
+def create_vpc_peering_dict(vpc_peering_info, peerings_enteries_dict, cluster_cidr):
+    if "providers" not in peerings_enteries_dict:
+        peerings_enteries_dict = {
+            "cidr_block": cluster_cidr,
+            "providers": [],
+        }
+    peerings_enteries_dict["providers"].append(vpc_peering_info)
+    return peerings_enteries_dict
+
+
+def find_cidr_duplicates_and_overlap(cluster_name: str, input_dict: dict):
+    cluster_cidr = input_dict["cidr_block"]
+    for provider in input_dict["providers"]:
+        account_vpc_list = []
+        account_vpc_mesh_list = []
+        vpc_peering_compare = provider.get("vpc_peering")
+        vpc_peering_cidr = vpc_peering_compare.get("cidr_block")
+        vpc_peering_name = vpc_peering_compare.get("vpc_name")
+        if provider.get("provider") == "account-vpc":
+            account_vpc_list.append(
+                (vpc_peering_name, ipaddress.ip_network(vpc_peering_cidr))
+            )
+        account_vpc_bool = vpc_peering_list_compare(account_vpc_list, cluster_name)
+        if provider.get("provider") == "account-vpc-mesh":
+            account_vpc_mesh_list.append(
+                (vpc_peering_name, ipaddress.ip_network(vpc_peering_cidr))
+            )
+        vpc_mesh_bool = vpc_peering_list_compare(account_vpc_mesh_list, cluster_name)
+        if (
+            provider.get("provider") == "cluster-vpc-requester"
+            or provider.get("provider") == "cluster-vpc-accepter"
+        ):
+            vpc_peering_compare = provider.get("vpc_peering")
+            if ipaddress.ip_network(cluster_cidr).overlaps(
+                ipaddress.ip_network(vpc_peering_compare["cidr_block"])
+            ):
+                logging.error(f"VPC peering error in cluster {cluster_name}")
+                logging.error(
+                    f"{cluster_name} overlaps with cidr block {vpc_peering_compare['cidr_block']}"
+                )
+                return False
+        if not account_vpc_bool or vpc_mesh_bool:
+            return False
+    return True
```

Comment:
As a result, we don't need to worry about which vpc CIDR comes from where. We just loop the entire peerings_enteries and find overlap.
```
    for i in range(len(peerings_enteries)):
        compared_vpc = peerings_enteries[i]
        for j in range(i + 1, len(peerings_enteries)):
            comparing_vpc = peerings_enteries[j]
            if ipaddress.ip_network(compared_vpc['cidr_block']).overlaps(
            comparing_vpc['cidr_block']
        ):
            logging.error(f"VPC peering error in cluster {cluster_name}")
            logging.error(
                f"vpc {compared_vpc['vpc_name']} with cidr block {compared_vpc['cidr_block']} provider by {compared_vpc['provider']} overlaps with vpc {comparing_vpc['vpc_name']} with cidr block {comparing_vpc['cidr_block']} provider by {comparing_vpc['provider']}"
            )
            return True
```
Since the name of the function is `find_cidr_duplicates_and_overlap`, we'd better return true when we find one.